---
title: "Practice_Workspace.Rmd"
author: "Jonah Merriam"
date: "8/10/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

The goal of this Rmd file is to create a reproducible set of code that will allow future people to repeat this analysis on 16S metagenomic data received from MRDNA. 

```{r Cleaning data in excel}
##The data given to me from my professor was from MR.DNA and was in an excel sheet. This excel sheet had different pages for the levels of phylum down to species. The one change I made to the excel sheet before reading it into R was to delete the empty columns in between the samples and in between the first sample and the label. This can be done in R but it took less than a minute in excel so I did it that way.
```


```{r packages}
##The first step to working through these steps in R will be to download these packages and make sure they are ready to use
install.packages("paletteer")
install.packages("janitor")
install.packages("kableExtra")
install.packages("skimr")
install.packages("here")
install.packages("readxl")
install.packages("tidyverse")
install.packages("vegan")
install.packages("reshape2")
install.packages("knitr")
install.packages("ggpubr")
```

``` {r libraries}
library(vegan)
library(tidyverse)
library(readxl)
library(here)
library(skimr) 
library(kableExtra)
library(janitor)
library(paletteer)
library(reshape2)
library(knitr)
library(ggpubr)
```

```{r Data Loading}
phylum_long<- read_excel(here("Frost IBS Data", "Phylum Long.xlsx"))
class_long<- read_excel(here("Frost IBS Data", "Class Long .xlsx"))
order_long<- read_excel(here("Frost IBS Data", "Order Long.xlsx"))
family_long<- read_excel(here("Frost IBS Data", "Family Long.xlsx"))
genus_long<- read_excel(here("Frost IBS Data", "Genus Long Alphabetical .xlsx"))
species_long<- read_excel(here("Frost IBS Data", "Species Long Alphabetical.xlsx"))

phylum_wide<- read_excel(here("Frost IBS Data", "Frost IBS Original Data.xlsx"))
class_wide<- read_excel(here("Frost IBS Data", "Class Wide.xlsx"))
order_wide<- read_excel(here("Frost IBS Data", "Order Wide.xlsx"))
family_wide<- read_excel(here("Frost IBS Data", "Family Wide.xlsx"))
genus_wide<- read_excel(here("Frost IBS Data", "Genus Wide.xlsx"))
species_wide<- read_excel(here("Frost IBS Data", "Species Wide.xlsx"))
```

```{r Cleaning Data}
options(scipen=999)
##Changes the scientific notation to normal numbers

##Cleaning the long form data
phylum_long_na<-read_excel((here("Frost IBS Data","Phylum Long.xlsx")), na="0")
##This code reads in the original data file and replaces all of the 0 values with NA which allows for easier cleaning

phylum_long_nona<-phylum_long_na[rowSums(is.na(phylum_long_na)) != ncol(phylum_long_na[3:3]), ]
##This will remove the rows of data that have only NA in the columns specified, in this case [3:3]

phylum_long_clean<- phylum_long_nona %>% clean_names() %>%  mutate(phylum = str_to_lower(phylum))
##This code will put the column titles into a standardized snake case and will change selected character columns to all lowercase so the data is easier to handle. 

##Cleaning wide data
order_long_na<-read_excel((here("Frost IBS Data","Order Wide.xlsx")), na="0")

order_long_nona<-order_long_na[rowSums(is.na(order_long_na)) != ncol(order_long_na[2:8]), ]

```

```{r Order Graph}
order_long<-read_excel(here("Frost IBS Data", "Order Long.xlsx"))

order_long_shortened<- order_long %>% filter(order_relative_abundance >= 0.5)

order_barchart_shortened<-ggplot(order_long_shortened, aes(y=order_relative_abundance,fill=order,x=name))+ geom_bar(stat='identity',color="black",aes(fill=order))+ scale_color_paletteer_d("awtools::bpalette")+ labs(title="", x="",y="Order Relative abundance above 0.5%")+theme_bw() + theme(legend.position="right",axis.text.x = element_text(angle=45, hjust=1,vjust=1,color="black"))

order_barchart_shortened
```

```{r Tables}
phylum_long<- read_excel(here("Frost IBS Data", "Phylum Long.xlsx"))
##Read in the long form data 

phylum_summary_table <- phylum_long %>% group_by(phylum) %>% summarize(mean_abundance = mean(phylum_relative_abundance), median_abundance = median(phylum_relative_abundance), sd_relative_abundance = sd(phylum_relative_abundance))
phylum_summary_table
##This table will sort the data for each individual into groups based on the phylum. It includes the mean, median, and standard deviation of each of the individuals relative abundances across each phylum.

phylum_table_above_1<- phylum_long %>% group_by(phylum) %>% filter(phylum_relative_abundance >= 1) %>% summarize(name = n(),mean_abundance = mean(phylum_relative_abundance), median_abundance = median(phylum_relative_abundance), sd_relative_abundance = sd(phylum_relative_abundance), shannon = (shannon_phylum))
kable(phylum_table_above_1)
##The filter on this allows it to weed out the phylum's with less than 1% relative abundance which gives a better idea of the most prevalent phylum's 
```

```{r Diversity calculations}
##Calculating Shannon with wide data set 

shannon_order<-diversity(order_wide[2:8],index="shannon",2)
shannon_order

shannon_phylum<-diversity(phylum_wide[2:8], index="shannon",2)
shannon_phylum

##Inverse Simpson
invsimp_phylum<-diversity(phylum_wide[2:8],index="invsimpson",2)

invsimp_order<-diversity(order_wide[2:8],index="invsimpson",2)

##Making a graphical representation
```

```{r Statistics}
options(scipen=999)
##Getting the data into a use able format by giving each separate group its own column so the tests can be run within a group 
order_stat<- t(order_wide)

##changes the column names to the first row
colnames(order_stat) <- as.character(order_stat[1,])

##gets rid of the first row 
order_stat <- order_stat[-c(1), ]
 
##turns it into data frame instead of collection of variables
order_stat<-data.frame(order_stat)

##Shifting all of the columns to numeric
order_stat<- sapply( order_stat, as.numeric )

##Getting rid of all columns with zero
order_stat<-order_stat[, !apply(order_stat == 0, 2, all)]

##Renaming the row names with the sample names


##Test for normal distribution with Shapiro-Wilkâ€™s test for normal distribution 
shapiro.test(order_stat[,c("acidimicrobiales")])
##This works for an individual column but I want to create a list for the whole data set where NA is listed if not enough samples and where the p-value can tell if it is normal and then have a data frame where only the normally distributed ones appear
order_shapiro_full<- shapiro.test((order_stat[,c[1:94]]))

```

